### DISCUSSION TITLE
[Feature] (More) Prometheus metrics

### DISCUSSION THREAD

**Discussion by joaopedrocg27** [immich-app/immich#11069] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I would like to have more metrics available like the number of photos, total library size, user activity, etc
Related to #10391

I found [this existing exporter](https://github.com/friendlyFriend4000/prometheus-immich-exporter), but I do not want to manage another service and take care of credentials, this could be built in to immich.

Does this make sense? I would be interested in developing it, just need some directions on here to start.

Example dashboard from the mentioned exporter:
![image](https://github.com/user-attachments/assets/3c11632c-2a29-4af5-a4fe-c0ce55f06c36)

Thank you guys, awesome project!

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
A PR would most certainly be welcome!

**Comment by jpds** [comment]
I find the Immich metrics endpoint as it currently stands in `1.232.3` ... bizarre. It has a series of metrics such as `immich_job_repository_queue_all_duration_count`, `immich_search_repository_search_places_duration_bucket`, `immich_user_repository_get_list_duration_bucket` etc. I do not know why these exist when `http_server_duration_bucket` seems to cover these with the appropriate labels instead of using these individual metric names.

Using the individual metric names; if I want to know which API endpoint is showing the highest latency - this is basically impossible to represent on a Grafana dashboard without making individual queries for all of these different names.

As stated in the original request; as an Immich server admin, I want to be able to answer this question in Prometheus:

1) How many users do I have in Immich? `immich_users_total 5`

2) How many photos do each of my users have in Immich, separated by label name? `immich_photos_total{user="fred"} 4230`

3) How much storage space are my users using individually in Immich? `immich_photos_bytes_total{user="fred"} 4758943753`

4) What quota does a user have set? `immich_user_quota_bytes_total{user="fred"} 5000000000`

Essentially, I want the `/admin/server-status` page but in a time-series database so that I can observe trends over time.

**Comment by tobiasff3200** [comment]
I would also like metrics about the background job queues. The graphs on the queue overview page are great, but it would be much easier to see them all in one Grafana dashboard. This would also enable alerts like unusual long or stuck queues.
